ghcide.cabal

```diff
@@ -123,6 +123,7 @@ library
         Development.IDE.LSP.Protocol
         Development.IDE.LSP.Server
         Development.IDE.Spans.Common
+        Development.IDE.Spans.Type
```

Comment:
This is the core change, moving `Development.IDE.Spans.Type` from `other-modules` to `exposed-modules`. This exposes the types defined in that module, enabling https://github.com/digital-asset/daml/pull/14792.